services/fee-estimator/shared/dynamicFees.js

```diff
@@ -34,14 +34,22 @@ const calculateBlockSize = async block => {
 };
 
 const calculateWeightedAvg = async blocks => {
-	const blockSizes = await Promise.all(blocks.map(block => calculateBlockSize(block)));
+	const blockSizes = await BluebirdPromise.map(
+		blocks, 
+		async block => calculateBlockSize(block),
+		{ concurrency : blocks.length},
+	);
 	const decayFactor = config.feeEstimates.wavgDecayPercentage / 100;
 	let weight = 1;
-	const wavgLastBlocks = blockSizes.reduce((a = 0, b = 0) => {
-		weight *= 1 - decayFactor;
-		return a + (b * weight);
-	});
+	const wSumLastBlocks = blockSizes.reduce(
+		(prevWSum = 0, blockSize) => {
+			const currentWSum = prevWSum + (blockSize * weight);
+			weight *= (1 - decayFactor);
+			return currentWSum;
+		},
+	);
 
+	const wavgLastBlocks = wSumLastBlocks / blocks.length;
 	return wavgLastBlocks;
```

Comment:
Please get the change confirmed with someone from the research team as well.
Also, please get your doubt clarified about the following logic in the `getEstimateFeeByteForBlock` method.
```
if (wavgBlockBatch > (12.5 * 2 ** 10) || sizeLastBlock > (14.8 * 2 ** 10)) {
```

Reply:
This comment is invalid!

Updating `const wavgLastBlocks = wSumLastBlocks / blocks.length;` to `const wavgLastBlocks = wSumLastBlocks / totalWeight;` clarifies it.